src/core/properties/listproperty.cpp

```diff
@@ -0,0 +1,240 @@
+    , uiFlags_(uiFlags)
+    , maxNumElements_("maxNumberOfElements", maxNumberOfElements) {}
+
+ListProperty::ListProperty(std::string identifier, const std::string& displayName,
+                           std::vector<std::unique_ptr<Property>>&& prefabs,
+                           size_t maxNumberOfElements, ListPropertyUIFlags uiFlags,
+                           InvalidationLevel invalidationLevel, PropertySemantics semantics)
+    : ListProperty(identifier, displayName, maxNumberOfElements, uiFlags, invalidationLevel,
+                   semantics) {
+    prefabs_ = std::move(prefabs);
+}
+
+ListProperty::ListProperty(std::string identifier, const std::string& displayName,
+                           std::unique_ptr<Property>&& prefab, size_t maxNumberOfElements,
+                           ListPropertyUIFlags uiFlags, InvalidationLevel invalidationLevel,
+                           PropertySemantics semantics)
+    : ListProperty(identifier, displayName, maxNumberOfElements, uiFlags, invalidationLevel,
+                   semantics) {
+    prefabs_.emplace_back(std::move(prefab));
+}
+
+ListProperty::ListProperty(const ListProperty& rhs)
+    : CompositeProperty(rhs)
+    , uiFlags_(rhs.uiFlags_)
+    , maxNumElements_(rhs.maxNumElements_)
+    , prefabs_(std::move(detail::clonePropertyVector(rhs.prefabs_))) {}
+
+ListProperty& ListProperty::operator=(const ListProperty& that) {
+    if (this != &that) {
+        uiFlags_ = that.uiFlags_;
+        maxNumElements_ = that.maxNumElements_;
+        prefabs_ = std::move(detail::clonePropertyVector(that.prefabs_));
+    }
+    return *this;
+}
+
+ListProperty* ListProperty::clone() const { return new ListProperty(*this); }
+
+std::string ListProperty::getClassIdentifierForWidget() const {
+    return ListProperty::CLASS_IDENTIFIER;
+}
+
+void ListProperty::set(const Property* src) {
+    if (const auto listproperty = dynamic_cast<const ListProperty*>(src)) {
+        set(listproperty);
+    }
+}
+
+void ListProperty::set(const ListProperty* src) {
+    NetworkLock l(this);
+    // check for matching prefab types first
+    if (getPrefabIDs() == src->getPrefabIDs()) {
+        // TODO: should we sync/consider the UI flags here as well?
+
+        maxNumElements_ = src->maxNumElements_;
+
+        while (size()) {
+            removeProperty(static_cast<size_t>(0));
+        }
+        // FIXME: cloning is not entirely appropriate
+        // e.g. does not consider observers, callbacks, and links
+        for (auto p : src->getProperties()) {
+            addProperty(p->clone(), true);
+        }
+
+        propertyModified();
+    } else {
+        LogWarn("ListProperty mismatch. Unable to link");
+    }
+}
+
+void ListProperty::setMaxNumberOfElements(size_t n) {
+    maxNumElements_ = n;
+    if (n > 0) {  // n = 0 means no limit
+        // remove superfluous list items
+        const bool removedElements = size() > n;
+        while (size() > n) {
+            removeProperty(size() - 1);
```

Comment:
Are these properties deleted?

Reply:
This calls the regular `removeProperty()` function which deletes owned properties (returns nullptr), or removes them from the owner and returns a pointer.

Reply:
Ok